### PR TITLE
Attempting to document --asymmetric-routes

### PR DESCRIPTION
**Pull request by stuclem**
Fixes https://github.com/vmware/vic-product/issues/210. Or, more accurately, attempts to.

Really not very sure about this, because networking really is a weak spot for me.

### CODE REVIEW

**Comment by stuclem** [comment]
@anchal-agrawal and @hmahmood can you please take a look? I anticipate that this will take a few drafts before we get it right.

File: docs/user_doc/vic_vsphere_admin/vch_installer_options.md [hunk 1]
```diff
@@ -570,6 +570,20 @@ If not specified, the VCH uses the public network for management traffic. If you
...
 <pre>--management-network <i>port_group_name</i></pre>
 
+### `--asymmetric-routes` <a id="asymmetric-routes"></a>
+
+Short name: `--ar`
+
+Allows incoming connections from ESXi hosts to VCHs over the public network rather than over the management network. If the management network does not have route entries for the vCenter Server and ESXi host subnets, containers can experience the following problems:
+
+-  Containers that run without specifying `-d` remain in the starting state.
+-  You can make connections to running containers via port forwarding from the public network, but not from the client or management networks.
```
Comment (hmahmood) [inline docs/user_doc/vic_vsphere_admin/vch_installer_options.md]: I don't think using this option would result in this working. Port bindings are only exposed on the public network.

File: docs/user_doc/vic_vsphere_admin/vch_installer_options.md [hunk 2]
```diff
@@ -570,6 +570,20 @@ If not specified, the VCH uses the public network for management traffic. If you
 
 <pre>--management-network <i>port_group_name</i></pre>
 
+### `--asymmetric-routes` <a id="asymmetric-routes"></a>
+
+Short name: `--ar`
+
+Allows incoming connections from ESXi hosts to VCHs over the public network rather than over the management network. If the management network does not have route entries for the vCenter Server and ESXi host subnets, containers can experience the following problems:
```
Comment (hmahmood) [inline docs/user_doc/vic_vsphere_admin/vch_installer_options.md]: This option was primarily added to allow containers on the bridge networks to access management/client assets via the public interface (i.e. indirectly), if they happen to be routable from the public network. I think you should add this explanation to the docs as well.

File: docs/user_doc/vic_vsphere_admin/vch_installer_options.md (hunk 1 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/vch_installer_options.md]: Deleted the second bullet.

File: docs/user_doc/vic_vsphere_admin/vch_installer_options.md (hunk 2 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/vch_installer_options.md]: Added a version of this in the description of `--asymmetric-routes`.

**Comment by stuclem** [comment]
Thanks @hmahmood. I incorporated your comments in https://github.com/vmware/vic-product/pull/728/commits/b1e2c8c637bd717fcb26869a51d4fa97c4126e4e. Is this OK now?

**Comment by stuclem** [comment]
One observation: since this is something that is likely to be discovered _after_ deployment of the VCH, shouldn't there be an `--asymmetric-routes` option for `vic-machine configure`?

File: docs/user_doc/vic_vsphere_admin/vch_installer_options.md [hunk 3]
```diff
@@ -570,6 +570,17 @@ If not specified, the VCH uses the public network for management traffic. If you
 
 <pre>--management-network <i>port_group_name</i></pre>
 
+### `--asymmetric-routes` <a id="asymmetric-routes"></a>
+
+Short name: `--ar`
+
+Allows incoming connections from ESXi hosts to VCHs over the public network rather than over the management network. This option allows containers on container networks to indirectly access assets on the management or client networks via the public interface, if those assets are routable from the public network. If the management network does not have route entries for the vCenter Server and ESXi host subnets,  and you do not set `--asymmetric-routes`, containers that run without specifying `-d` remain in the starting state.
```
Comment (hmahmood) [inline docs/user_doc/vic_vsphere_admin/vch_installer_options.md]: > This option allows containers on container networks

Actually only containers on bridge networks.

**Comment by hmahmood** [comment]
@stuclem opened an issue for configure: https://github.com/vmware/vic/issues/6225

**Comment by stuclem** [comment]
Thanks @hmahmood. I fixed your bridge network comment in https://github.com/vmware/vic-product/pull/728/commits/3f33425a43e7c7badc13a772ed50ea6af36ec205.

**Pull request merged**